src/components/welcome/MyAccounts/AccountItem.tsx

```diff
@@ -90,8 +130,8 @@ const AccountItem = ({ onLinkClick, safeItem }: AccountItemProps) => {
           <Box pr={2.5}>
             <SafeIcon
               address={address}
-              owners={safeOverview?.owners.length ?? counterfactualSetup?.owners.length}
-              threshold={safeOverview?.threshold ?? counterfactualSetup?.threshold}
+              owners={safeOwners.length ?? undefined}
```

Comment:
`safeOwners` is always an array now so `safeOwners.length` could resolve to 0 and fail the check. We should instead check `safeOwners.length > 0 ? safeOwners.length : undefined` in order to keep the same behaviour.